zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java

```diff
@@ -46,16 +43,19 @@
   public static final AttributeKey<String> NAME = AttributeKey.valueOf("name");
 
   public interface BodyConverter<V> {
-    /** Most convert with {@link HttpData#toStringUtf8()} or {@link #toInputStream(HttpData)} */
+    /** The source is from {@link AggregatedHttpResponse}, so act accordingly. */
     V convert(HttpData content) throws IOException;
+  }
 
-    /** Use this when you don't need a string or only need to read the response once. */
-    // TODO: once https://github.com/line/armeria/issues/1918 is done, switch back to an interface
-    default InputStream toInputStream(HttpData content) {
-      if (content instanceof ByteBufHolder) {
-        return new ByteBufInputStream(((ByteBufHolder) content).content());
-      } else {
-        return content.toInputStream();
+  public interface InputStreamConverter<V> extends BodyConverter<V> {
+    V convert(InputStream content) throws IOException;
+
+    @Override default V convert(HttpData content) throws IOException {
```

Comment:
Wondering, what do you think about this

```java
V convert(JsonParser parser, Supplier<String> contentString);

@Override default V convert(HttpData content) throws IOException {
  try (InputStream stream = content.toInputStream()) {
    return convert(JSON_FACTORY.createParser(stream), content::toStringUtf8);
  } finally {
    ReferenceCountUtil.safeRelease(content);
  }
}
```

I think all our conversions are of JSON so can pass `JsonParser` (or leave it as `InputStream` anyways), and it also allows the converters to have access to the full content string if they want it for an error message or something. Maybe it's too complex (a bit hesitant to make suggestions that add any complexity at this point...).

Reply:
will think about it. right now I need to unwind the client which is unrelated and get back to this after.. thanks for the suggestion!

Reply:
I think I get this.. I've a nag in head to help weave this in after getting aws working. mainly an issue of drift control at the moment